broker/lib/controllers/ServiceBrokerApiController.js

```diff
@@ -131,7 +266,59 @@ class ServiceBrokerApiController extends FabrikBaseController {
       .catch(ServiceInstanceNotFound, gone);
   }
 
+  deleteInstanceV2(req, res) {
+    const params = _.omit(req.query, 'plan_id', 'service_id');
+
+    function done(result) {
+      let statusCode = CONST.HTTP_STATUS_CODE.OK;
+      const body = {};
+      if (req.instance.async) {
+        statusCode = CONST.HTTP_STATUS_CODE.ACCEPTED;
+        body.operation = utils.encodeBase64(result);
+      }
+      res.status(statusCode).send(body);
+    }
+
+    function gone(err) {
+      /* jshint unused:false */
+      res.status(CONST.HTTP_STATUS_CODE.GONE).send({});
+    }
+    req.operation_type = CONST.OPERATION_TYPE.DELETE;
+    this.validateRequest(req, res);
+    return Promise.try(() => {
+        if (req.manager.name === CONST.INSTANCE_TYPE.DIRECTOR) {
+          // Acquire lock for this instance
+          return lockManager.lock(req.params.instance_id, {
+            lockType: CONST.ETCD.LOCK_TYPE.WRITE,
+            lockedResourceDetails: {
+              resourceType: CONST.APISERVER.RESOURCE_TYPES.DEPLOYMENT,
+              resourceName: CONST.APISERVER.RESOURCE_NAMES.DIRECTOR,
+              resourceId: req.params.instance_id,
+              operation: CONST.OPERATION_TYPE.DELETE
+            }
+          });
+        }
+      })
+      .then(() => req.instance.delete(params))
+      .then(done)
+      .catch(err => {
+        if (err instanceof EtcdLockError) {
+          throw err;
+        }
+        return lockManager.unlock(req.params.instance_id)
+          .throw(err);
+      })
+      .catch(ServiceInstanceNotFound, gone);
+  }
```

Comment:
Since all put, delete and patch have almost same options provided in lock function like [here](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/304/files#diff-03e33ae33c13c68e445aa8cc1048acc9R291), for more readability, we can call a common function to get lock  details. Also for [lastoperation lock](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/304/files#diff-03e33ae33c13c68e445aa8cc1048acc9R399).

Reply:
Fixed as locking now happens by middleware.